File: 02-programacion-basica/08-arreglos-y-matrices/java/soluciones/RetoAnalizarNotas.java
public class RetoAnalizarNotas {
    public static void main(String[] args) {
        int[] notas = {15, 18, 9, 20, 12};
        int suma = 0;
        int mayor = notas[0];
        int menor = notas[0];
        int aprobados = 0;

        for (int nota : notas) {
            suma += nota;
            if (nota > mayor) mayor = nota;
            if (nota < menor) menor = nota;
            if (nota >= 11) aprobados++;
        }

        // El casting evita una división entera al calcular el promedio.
        double promedio = (double) suma / notas.length;
        System.out.printf("Promedio: %.2f | Mayor: %d | Menor: %d | Aprobados: %d%n",
                promedio, mayor, menor, aprobados);
    }
}

File: 02-programacion-basica/08-arreglos-y-matrices/scala/soluciones/RetoAnalizarNotas.scala
@main def retoAnalizarNotas(): Unit =
  val notas = Array(15, 18, 9, 20, 12)
  val promedio = notas.sum.toDouble / notas.length
  val mayor = notas.max
  val menor = notas.min
  // count calcula cuántos elementos satisfacen la condición.
  val aprobados = notas.count(_ >= 11)
  println(f"Promedio: $promedio%.2f | Mayor: $mayor | Menor: $menor | Aprobados: $aprobados")
